udf/mrn_udf_highlight_html.cpp

```diff
@@ -415,12 +445,62 @@ MRN_API char *mroonga_highlight_html(UDF_INIT *init,
   *is_null = 0;
   GRN_BULK_REWIND(&(info->result));
 
-  if (!highlight_html(ctx,
-                      reinterpret_cast<grn_pat *>(keywords),
-                      args->args[0],
-                      args->lengths[0],
-                      &(info->result))) {
-    goto error;
+  if (info->target_type == HTML) {
+    const char *previous_position = args->args[0];
+    const char *end_position = args->args[0] + args->lengths[0];
+    const char *current_position = args->args[0];
+    bool is_in_tag = false;
+
+    while (current_position < end_position) {
+      current_position++;
```

Comment:
We should not assume a character is one byte.

We can use `grn_charlen()` to count a character size in bytes.

Reply:
Fixed.